### PR TITLE
[ADD] estate: Create custom views T#72976

### PR DESCRIPTION
**Pull request by gerashdo**
**Chapter 7**
- Create custom list view with specific fields to display.
- Create custom form view with custom layout and fields organization.
- Create search view with custom filter and group by options.
- Exercises reference link: https://www.odoo.com/documentation/16.0/developer/tutorials/getting_started/07_basicviews.html#

**Chapter 8**
- Create `estate.property.type` model, also its menu, action and custom form view.
- Exercise reference link: https://www.odoo.com/documentation/16.0/developer/tutorials/getting_started/08_relations.html#many2one

### CODE REVIEW

**Comment by gerashdo** [comment]
Ready for review @JesusValdez96 🙂.

File: estate/models/estate_property_type.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from odoo import fields, models
+
+
+class PropertyType(models.Model):
+    _name = "estate.property.type"
+    _description = "Type of a property"
+
+    name = fields.Char()
+
```
Comment (JesusValdez96) [inline estate/models/estate_property_type.py]: Did you run pre-commit? 

Add a blank line at the end of file.

File: estate/views/estate_property_type_views.xml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<?xml version='1.0' encoding='utf-8'?>
```
Comment (JesusValdez96) [inline estate/views/estate_property_type_views.xml]:
```suggestion
<?xml version='1.0' encoding='utf-8' ?>
```

File: estate/views/estate_property_type_views.xml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+                </sheet>
+            </form>
+        </field>
+    </record>
+
+    <record id="estate_property_type_action" model="ir.actions.act_window">
+        <field name="name">Property Types</field>
+        <field name="res_model">estate.property.type</field>
+        <field name="view_mode">tree,form</field>
+    </record>
```
Comment (JesusValdez96) [inline estate/views/estate_property_type_views.xml]: This will be autofixed by `pre-commit-vauxoo -t all`

```suggestion
    </record>
```

**Review by JesusValdez96 — CHANGES_REQUESTED** [review]
Good job @gerashdo only run pre-commit and it's done.

I forgot to tell you that you can install these useful extensions:

- eamodio.gitlens
- ms-python.pylint
- dotjoshjohnson.xml
- ms-python.python
-  ms-python.vscode-pylance
-  ms-python.flake8
- dbaeumer.vscode-eslint
-  redhat.vscode-xml
- shardulm94.trailing-spaces

File: estate/models/estate_property_type.py (hunk 1 above)
Comment (gerashdo) [reply estate/models/estate_property_type.py]: I remember that I did, but surly I did not 😅.
Thank you!

**Comment by gerashdo** [comment]
@JesusValdez96  I am not sure what is happening here.
I run `pre-commit-vauxoo -t all`, and it leaves a blank line, if I run it again everything is correct but, when I push the changes the blank line is not in the remote repository.

**Local**
image

**Remote**
image

**Comment by JesusValdez96** [comment]
@gerashdo in remote it's ok, when there is no new line at the end, you will see an icon like this
newline

**Review by JesusValdez96 — APPROVED** [review]

**Comment by gerashdo** [comment]
Can it be merged now, or there are pending changes? @JesusValdez96

**Pull request merged**